code-challenges.test.js

```diff
@@ -14,39 +14,115 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
-// b) Create the function that makes the test pass.
+
+describe("codedMessage", () => {
+    it ("takes in a string and returns a string with a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+//   ReferenceError: codedMessage is not defined
+
+
+
+// Create a function called 'codedMessage' that takes in one parameter: a string.
+const codedMessage = (string) => {
+    // Iterate through the string to find the vowels, a, e, i, o and replace them with the specified number.  
+    for (let i=0; i<string.length; i++)
+    // We want to make sure the letters are case insensitive, so we can yield all proper results, so we'll use the Regex method i when we replace ALL the vowels (not just the first vowel) with a number.
+    return string.replaceAll(/['a']/gi, 4).replaceAll(/['e']/gi, 3).replaceAll(/['i']/gi, 1).replaceAll(/['o']/gi, 0)
```

Comment:
Logic looks great!  One thing to consider is using replaceAll combined with Regex will automatically iterate eliminating the need for a for loop.